python/ray/data/datasource/file_based_datasource.py

```diff
@@ -371,11 +372,19 @@ def __init__(
         self._partition_filter = partition_filter
         self._partitioning = partitioning
         self._block_udf = _block_udf
+        self._ignore_missing_paths = ignore_missing_paths
         self._reader_args = reader_args
         paths, self._filesystem = _resolve_paths_and_filesystem(paths, filesystem)
         self._paths, self._file_sizes = meta_provider.expand_paths(
-            paths, self._filesystem
+            paths, self._filesystem, ignore_missing_paths=ignore_missing_paths
         )
+
+        if ignore_missing_paths and len(self._paths) <= 0:
```

Comment:
Nit: Length of a list can never be negative, so no need to be defensive here!
```suggestion
        if ignore_missing_paths and len(self._paths) == 0:
```

Reply:
added